dataset/src/main/scala/frameless/ops/RelationalGroupsOps.scala

```diff
@@ -0,0 +1,212 @@
+package frameless
+package ops
+
+import org.apache.spark.sql.catalyst.analysis.UnresolvedAlias
+import org.apache.spark.sql.catalyst.plans.logical.{MapGroups, Project}
+import org.apache.spark.sql.{Column, Dataset, FramelessInternals, RelationalGroupedDataset}
+import shapeless.ops.hlist.{Mapped, Prepend, ToTraversable, Tupler}
+import shapeless.{::, HList, HNil, ProductArgs}
+
+/**
+  * @param groupingFunc functions used to group elements, can be cube or rollup
+  * @tparam T the original `TypedDataset's` type T
+  * @tparam TK all columns chosen for aggregation
+  * @tparam K individual columns' types as HList
+  * @tparam KT individual columns' types as Tuple
+  */
+private[ops] abstract class RelationalGroupsOps[T, TK <: HList, K <: HList, KT]
```

Comment:
Do you think it's possible to remove `GroupedByManyOps` and use this class instead? The implementations look vers similar.

Reply:
Sounds good to me!

Reply:
Okay, I have moved quite a bit of repeating code to `AggregatingOps` which is extended by both `GroupedByManyOps` and `RelationalGroupsOps` but maybe there is a better way